### PR TITLE
Docs update to provide more IAM information on aws.md

### PR DESCRIPTION
**Pull request by jasondewitt**
Updated aws.md to give more info about IAM, the link previously went to SQS documentation. I also added some details about how to configure your kops IAM user that were posted by @azell in #1103

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kops/1493)

### CODE REVIEW

**Comment by jasondewitt** [comment]
i signed it!

**Comment by chrislovecnm** [comment]
@k8s-bot ok to test

File: docs/aws.md [hunk 1]
```diff
@@ -47,10 +47,32 @@ brew install kubernetes-cli
 ### Setting up a kops IAM user
 
 
-In this example we will be using a dedicated IAM user to use with kops. This user will need basic API security credentials in order to use kops. Create the user and credentials using the AWS console. [More information](http://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSGettingStartedGuide/AWSCredentials.html).
+In this example we will be using a dedicated IAM user to use with kops. This user will need basic API security credentials in order to use kops. Create the user and credentials using the AWS console. [More information](https://aws.amazon.com/documentation/iam/).
 
 Kubernetes kops uses the official AWS Go SDK, so all we need to do here is set up your system to use the official AWS supported methods of registering security credentials defined [here](https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/configuring-sdk.html#specifying-credentials). Here is an example using the aws command line tool to set up your security credentials.
 
+To create a IAM user for kops do the following:
```
Comment (kris-nova) [inline docs/aws.md]: This PR is fabulous.. thank you for your efforts!

So the rest of the documentation offers reproducible code snippets that are generally formatted using bash syntax highlighting. Can we also offer a copy/paste code snippet for creating the IAM user from the command line? It lessons the error margin for interpretation

File: docs/aws.md [hunk 2]
```diff
@@ -47,10 +47,32 @@ brew install kubernetes-cli
...
+ - AmazonVPCFullAccess
+* Click Create Group
+* Click the Users link on the left tab and then click Add user
+* Set User name to kops
+* Select Access type Programmatic access
+* Click Next: Permissions
+* Select the kops Group and click Next: Review
+* Click Create user
+* Click on Download .csv
+* Run aws configure
```
Comment (kris-nova) [inline docs/aws.md]: Can we move this into a code block with syntax highlighting?

File: docs/aws.md [hunk 3]
```diff
@@ -47,10 +47,32 @@ brew install kubernetes-cli
...
+* Run aws iam list-users
```
Comment (kris-nova) [inline docs/aws.md]: Code block here too please

**Comment by jasondewitt** [comment]
Thanks for the feedback, I'll make these changes and resubmit.

Do we want to drop the AWS console instructions all together and go with awscli commands? or have both? I feel like keeping the list of required IAM policies and then giving the awscli commands would be the most informative and least error prone.

**Comment by jasondewitt** [comment]
Updated the PR based on Kris' requested changes above. While making these updates I came upon a point of confusion, I was giving instructions to create the user via awscli *before* we told the user to install/configure awscli. I tried to clarify this a bit in the text, instructing them to either create the kops user via the AWS console, or if they already have a functioning awscli, use the bash commands. Then reconfigure awscli to use the security creds for the newly created kops user.

**Comment by chrislovecnm** [comment]
@kris-nova assigned to you and @jasondewitt has questions

File: docs/aws.md [hunk 4]
```diff
@@ -44,16 +44,44 @@ brew install kubernetes-cli
...
+The kops user will require the following IAM permission to function properly:
+```
+AmazonEC2FullAccess
+AmazonRoute53FullAccess
+AmazonS3FullAccess
+IAMFullAccess
+AmazonVPCFullAccess
+```
+
+If you already have a functioning AWS cli envrionment, you can create the kops user by doing the following:
```
Comment (kris-nova) [inline docs/aws.md]: Can we move this block below `Testing your DNS setup` and above `Setting up a state store for your cluster?` The way it reads we are asking users to use the `aws` cli before we tell them how to install it :)

File: docs/aws.md [hunk 5]
```diff
@@ -44,16 +44,44 @@ brew install kubernetes-cli
 
 ## Setup your environment
 
-### Setting up a kops IAM user
+## Create an IAM user for kops
```
Comment (kris-nova) [inline docs/aws.md]: Can we move this block below `Testing your DNS setup` and above `Setting up a state store for your cluster?` The way it reads we are asking users to use the `aws` cli before we tell them how to install it :)

Comment (jasondewitt) [inline docs/aws.md]: yeah, i had that same issue with the way it reads, wasn't sure of the best place to move this stuff to.

I'll make that change and have it pushed momentarily.

File: docs/aws.md [hunk 6]
```diff
@@ -148,8 +141,42 @@ subdomain.kubernetes.com.        172800  IN  NS  ns-1022.awsdns-35.com.
...
+The kops user will require the following IAM permission to function properly:
+```
+AmazonEC2FullAccess
+AmazonRoute53FullAccess
+AmazonS3FullAccess
+IAMFullAccess
+AmazonVPCFullAccess
+```
+
+If you already have a functioning AWS cli envrionment, you can create the kops user by doing the following:
```
Comment (kris-nova) [inline docs/aws.md]: Suggest:

```
You can create a kops IAM user from the command line using the following:
```

File: docs/aws.md [hunk 7]
```diff
@@ -148,8 +141,42 @@ subdomain.kubernetes.com.        172800  IN  NS  ns-1022.awsdns-35.com.
...
+arn:aws:iam::aws:policy/AmazonVPCFullAccess"
+
+for arn in $arns; do aws iam attach-group-policy --policy-arn "$arn" --group-name kops; done
+
+aws iam create-user --user-name testuser
+
+aws iam add-user-to-group --user-name kops --group-name kops
+
+aws iam create-access-key --user-name kops
+# make not of the SecretAccesKey and AccessKeyID in the returned json output
```
Comment (kris-nova) [inline docs/aws.md]:
```
s/not/note
```

**Comment by justinsb** [comment]
LGTM, by the way :-)  Thanks!

File: docs/aws.md [hunk 8]
```diff
@@ -148,8 +141,42 @@ subdomain.kubernetes.com.        172800  IN  NS  ns-1022.awsdns-35.com.
...
+for arn in $arns; do aws iam attach-group-policy --policy-arn "$arn" --group-name kops; done
+
+aws iam create-user --user-name testuser
+
+aws iam add-user-to-group --user-name kops --group-name kops
+
+aws iam create-access-key --user-name kops
+# make note of the SecretAccesKey and AccessKeyID in the returned json output
+```
+Kubernetes kops uses the official AWS Go SDK, so all we need to do here is set up your system to use the official AWS supported methods of registering security credentials defined [here](https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/configuring-sdk.html#specifying-credentials). Follow the example below to configure the aws command line tool to set up the security credentials for the kops user you just created.
```
Comment (kris-nova) [inline docs/aws.md]: This paragraph seems odd. We already suggest that the user runs `aws configure` a few lines above.. 

Also this paragraphs suggests that the user needs to "follow the example below" but I am not seeing any example below the paragraph 😞

Comment (jasondewitt) [inline docs/aws.md]: you are right, that line should have been removed when I moved the kops IAM user section to it's new location. Just pushed a change to fix that.

**Comment by jasondewitt** [comment]
ok i'm confused where the conflict is coming from on this now...

**Comment by chrislovecnm** [comment]
@k8s-bot aws e2e test this

**Comment by chrislovecnm** [comment]
I am guessing the testing problem is a flake. Travis CI has also been flaky as well. A flake is a false positive during testing.

**Comment by chrislovecnm** [comment]
I closed and reopened to bounce CI and e2e.

**Comment by chrislovecnm** [comment]
Can you do a rebase? I am wondering if we are flaking or the master when you branched had e2e issues

**Comment by jasondewitt** [comment]
sadly, i'm getting a little out of my git skills range with the conflict issue, I've never worked on a repo that moves this fast with so many contributors =). I attempted a rebase and I've ended up with my feature work in master on my fork and ruined my copy of master over on that fork.

It might be best/easiest if we close out this PR and I create a new fork, re-integrate my completed changes on a fresh copy of master and submit a new PR. If that is acceptable to you guys.

**Comment by kris-nova** [comment]
@jasondewitt If you think it's best to open a PR then please gopher it! Just close this one, and ping me.. this change is good on my end and I am ready to click merge as soon as we get a healthy looking PR.. 

Also feel free to ping me on slack if you are interested in walking through a rebase.. I have a script that makes it fairly easy.. and I am happy to help get this one doctored up as well.. Just let us know what you decide.

Cheers

**Pull request closed without merging**